### ISSUE TITLE
Undo fails with source editor open

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#2258]
1. Create new HTML file
2. Drag/drop some widgets (e.g., Button, Calendar, TextBox)
3. Open up the split source view
4. Click undo and see:

```
Uncaught Error: NOT_FOUND_ERR: DOM Exception 8
_1010.removeChilddavinci.js:19497
_12f8.undodavinci.js:23379
_19d3.undodavinci.js:34067
dojo.withDocdojo.js:5861
_1806.undodavinci.js:31313
_1b3d.rundavinci.js:36252
_3bb._runActiondavinci.js:4638
_37edojo.js:4535
_17cf._onClickdavinci.js:31011
_25edojo.js:2636
_25db._onClickdavinci.js:53787
_383dojo.js:4551
handles_OnDijitClickMixin.js:83
```

**Comment by JonFerraiolo** [comment]
-> @peller , although if @aerwin wants to take a crack at this, go ahead and reassign to yourself

@jhpedemonte @childsb @billreed63

**Comment by peller** [comment]
There seems to be some inconsistency with widgetIds on the dijits between the original instantiation and when the page is reparsed, making the undo stack invalid.  For example, drop a Calendar widget on the page, and widgetid="Calendar_0"  Reparse the page, and it's "input_1".  The undo command looks up the widget id "Calendar_0" and finds the stale widget and tries to remove it from the document, but it's already removed (wonder why it wasn't also destroyed)

**Comment by peller** [comment]
@JonFerraiolo @jhpedemonte 
Drop a button on a new page.
Context.getUniqueId calculates an id of "Button_0" based on idRoot=="Button" passed in by createWidget, based on metadata.
Change to split view.  setSource is triggered which reparses the doc.
Context.getUniqueId is called again.  This time, idRoot is undefined by the caller (visit) and node.tag is "input" so the id is "input_0"

**Comment by jhpedemonte** [comment]
I assume the issue here is that we're getting the "node" from the VE DOM, rather than the Model.  Since Dojo (and other widgets) can replace the existing DOM nodes with something entirely different, we **cannot depend on the VE DOM** for these things, and should **use the model** instead.

**Comment by JonFerraiolo** [comment]
Reopening after reverting fix with c466fa0995 . We need to take a different approach to fixing this problem. The fix in c466fa0995 caused all widgets to use their tagname as their ID. This means that all mobile views get the name "div_0", "div_1", ..., versus what we had in Preview 5 of "ScrollableView_0", "ScrollableView_1", etc. @jhpedemonte and @JonFerraiolo discussed this and both felt strongly that their usability advantages of "ScrollableView_0", etc, are too important.

**Comment by peller** [comment]
Github didn't seem to pick up on the checkin.  The regression started when getUniqueId started taking 3 parameters: 09b64b52d6b6bb287ce357d416e34b2cf14e4718  If the id is not persistent, the base algorithm should be used so the results will be consistent.  Changed the API to only take an id and an optional idRoot: 1c53ceb033649095b735b671fcb27bfaec57f15f
